django_forms_encoder/tests/test_encoder.py

```diff
@@ -0,0 +1,144 @@
+import json
+
+from django import forms
+from django.test import SimpleTestCase
+
+from django_forms_encoder import DjangoFormsEncoder
+
+
+class TestForm(forms.Form):
+    first_name = forms.CharField(required=True, label="First name")
+    last_name = forms.CharField(required=False, label="Last name")
+    email = forms.CharField(widget=forms.EmailInput())
+    age = forms.IntegerField(required=False, help_text="Should be at least 18")
+    is_admin = forms.BooleanField(required=False)
+    cv = forms.CharField(required=False, widget=forms.Textarea())
+
+    def clean(self):
+        raise forms.ValidationError('My hovercraft is full of eels!', code='invalid')
+
+
+class TestDjangoFormsEncoder(SimpleTestCase):
+    """Test django forms encoder."""
+
+    def test_encode_unbound_form(self):
+        self.assertEqual(json.loads(json.dumps(TestForm(), cls=DjangoFormsEncoder)), {
+            'fields': [{
+                'name': 'first_name',
+                'label': 'First name',
+                'required': True,
+                'widget': {
+                    'name': 'input',
+                    'type': 'text',
+                },
+            }, {
+                'name': 'last_name',
+                'label': 'Last name',
+                'required': False,
+                'widget': {
+                    'name': 'input',
+                    'type': 'text',
+                },
+            }, {
+                'name': 'email',
+                'label': 'email',
+                'required': True,
+                'widget': {
+                    'name': 'input',
+                    'type': 'email',
+                },
+            }, {
+                'name': 'age',
+                'label': 'age',
+                'required': False,
+                'widget': {
+                    'name': 'input',
+                    'type': 'number',
+                },
+                'help_text': 'Should be at least 18',
+            }, {
+                'name': 'is_admin',
+                'label': 'is_admin',
+                'required': False,
+                'widget': {
+                    'name': 'input',
+                    'type': 'checkbox',
+                },
+            }, {
+                'name': 'cv',
+                'label': 'cv',
+                'required': False,
+                'widget': {
+                    'name': 'textarea',
+                },
+            }]
+        })
```

Comment:
I'd rather see this split into several tests, one for each feature of the encoder.

Reply:
I tried to do that, but it leads to comparing long lists of inidividual properties, such as:

```python
self.assertEqual(
    [field['label'] for field in self.encoded_form['fields']],
    ['First name', 'Last name', 'email', 'age', 'is_admin', 'cv']
)
```

I feel that this is much less comprehensible. So, unless you have an idea how to split the tests and still keep them clear and easy to understand, I'll keep them as they are.

Reply:
That is caused by the single large `TestForm`. I'd rather split that as well to separate small test forms in each test. Any change in the `TestForm` may now lead to several changes in unrelated tests, which complicates any subsequent changes.

Reply:
Yeah, but that means having separate `Form` in each testcase, which makes the tests quite long and (again) less comprehensible.

Reply:
I've completely overhauled encoder tests. I agree that it is more legible now, even if a little bit longer.

Reply:
:+1: